common/src/main/java/com/microsoft/identity/common/internal/ui/webview/ISendResultCallback.java

```diff
@@ -0,0 +1,34 @@
+// Copyright (c) Microsoft Corporation.
+// All rights reserved.
+//
+// This code is licensed under the MIT License.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+package com.microsoft.identity.common.internal.ui.webview;
+
+/**
+ * Should send a form of a result when invoked.
+ */
+public interface ISendResultCallback {
+
+    /**
+     * Runs logic that sends a form of a result.
+     */
+    void onResultReady();
```

Comment:
Should this accept an input result? What result does it send and where?

Reply:
This callback is a part of this portion of my description: 
Made changes to WebViewAuthorizationFragment and AzureActiveDirectoryWebViewClient to set a callback that sends the auth result and finishes (in order to delay sending results until the user removes their smartcard). This callback will be invoked immediately if the cba challenge handler is null, and otherwise will be invoked based on the status of the challenge handler. A callback is needed here in order to make sure the user removes their smartcard before our library's activity destroys.

I needed a way to delay the sendResult() and finish() method calls within WebViewAuthorizationFragment. [This line](https://github.com/AzureAD/microsoft-authentication-library-common-for-android/pull/1966/files#diff-82c6ecf5054ca1eddf5258b0521e7162c60d136c8b1d7dc1d0003658a74ee852R300) is where the callback is first created, and it gets passed down to the AzureActiveDirectoryWebViewClient instance, which passes it to the challenge handler for CBA.

Reply:
(Any suggestions for method renames, preferably not tied to CBA, are appreciated)